packages/back-end/src/util/secrets.ts

```diff
@@ -22,19 +22,38 @@ export const UPLOAD_METHOD = (() => {
   return "local";
 })();
 
-export let MONGODB_URI =
-  process.env.MONGODB_URI ??
-  (prod ? "" : "mongodb://root:password@localhost:27017/test?authSource=admin");
+let MONGODB_URI = process.env.MONGODB_URI;
+
 if (!MONGODB_URI) {
-  throw new Error("Missing MONGODB_URI environment variable");
+  // Check for alternate mongo db environment variables
+  if (
+    process.env.MONGODB_USERNAME &&
+    process.env.MONGODB_PASSWORD &&
+    process.env.MONGODB_HOSTNAME &&
+    process.env.MONGODB_DBNAME
+  ) {
+    MONGODB_URI = `mongodb://${process.env.MONGODB_USERNAME}:${process.env.MONGODB_PASSWORD}@${process.env.MONGODB_HOSTNAME}/${process.env.MONGODB_DBNAME}`;
+
+    // Add extra args if they exist
+    if (process.env.MONGODB_EXTRA_ARGS) {
+      MONGODB_URI += `?${process.env.MONGODB_EXTRA_ARGS}`;
+    }
+  }
 }
 
 // For backwards compatibility, if no dbname is explicitly set, use "test" and add the authSource db.
 // This matches the default behavior of the MongoDB driver 3.X, which changed when we updated to 4.X
-if (MONGODB_URI.match(/:27017(\/)?$/)) {
```

Comment:
This logic needs to remain for the case that MONGODB_URI = `mongodb://root:password@localhost:27017/`.  I would suggest the following logic before this if:
```
if(!MONOGODB_URI) {
   if (prod) {
      throw new Error("Missing MONGODB_URI or required alternate environment variables to generate it");
   }
  MONGODB_URI = "mongodb://root:password@localhost:27017/test?authSource=admin"
}
```